### DISCUSSION TITLE
Having the possibility to disable layouts

### DISCUSSION THREAD

**Discussion by janwillemvd** [directus/directus#13883] [Feature Requests]
We'd like to have the ability to disable some of the lay-outs a user can choose in the 'Layout options' panel for a certain collection:

![Screenshot 2022-06-14 at 13 12 58](https://user-images.githubusercontent.com/10515908/173564536-b1d781a3-f604-4c9c-ab4b-8730b0daff01.png)

For instance, when you have a collection which doesn't have any location aware fields, a 'Map' layout makes no sense. Or a collection without any date aware fields probably doesn't need a 'Calendar' layout.

I think, we can do two things here:

- make the available layouts configurable per collection in the Data model section (for all users, regardless of the user and/or user role)
- make the available layouts configurable per collection per user role in the 'Roles and permissions' section

Personally, I think the first option would be the most easy one and this would be enough for me.

Any thoughts on this?

**Comment by benhaynes** [comment]
I agree with this.

First, one option to keep config simple is to introspect the collection's available fields (based on current user permissions) and then show/hide Layouts based on the correct types being available.

I am also wondering if it should be combined into the **Settings > Presets**. Presets is slated to get a major overhaul, and it supports user, role, and global options around Collection Detail (Layout) config.

Another option would be to create a _new_ section within **Settings > Data Model > Collection** for Layouts. Clean, makes sense.

Finally, I've had the idea that there should be **Settings > Layouts** for a while. Currently, there is no where to configure the layouts except at the user level. So if you needed to add an API Key for a Map Layout (for instance), that would give us a place to configure that... but I don't know how much this helps with the question you raised.

**Reply by janwillemvd** [reply]
Thanks for the quick response @benhaynes!

> Presets is slated to get a major overhaul, and it supports user, role, and global options around Collection Detail (Layout) config.

Nice!

> Another option would be to create a _new_ section within **Settings > Data Model > Collection** for Layouts. Clean, makes sense.

I would be happy to create a PR for this?

**Reply by benhaynes** [reply]
Thank you! Let's see which of these directions @rijkvanzanten prefers... and then I'm sure he'd _love_ to coordinate a PR with you!

**Reply by rijkvanzanten** [reply]
I don't think hiding some of the layouts in that little dropdown is worth the overhead of a whole new field in collections + the potential breaking change shortly to replace it with a bigger solution. Feels like it doesn't meet 80/20 to me. That being said, hiding layouts that rely on a field of a particular data type makes total sense to me. It's pointless to select the map layout if you don't have any geospatial data in your collection for example.

**Comment by janwillemvd** [comment]
> That being said, hiding layouts that rely on a field of a particular data type makes total sense to me. It's pointless to select the map layout if you don't have any geospatial data in your collection for example.

Yes, exactly!

So, then it would be something like:

- Table: always useful
- Cards: only useful when having images 
- Map: only useful when having geospatial data 
- Calendar: only useful when having date(time) data

?

**Reply by rijkvanzanten** [reply]
Sounds correct to me! The cards one is a little opinionated, as you could _theoretically_ still use it without an image (and just use the fallback icon), but at that point you kinda lose it's usefulness 🤔

**Reply by janwillemvd** [reply]
I agree, it then looses it's usefulness.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
